packages/roosterjs-content-model-plugins/lib/pluginUtils/splitTextSegment.ts

```diff
@@ -0,0 +1,42 @@
+import { createText } from 'roosterjs-content-model-dom';
+import type { ContentModelParagraph, ContentModelText } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export function splitTextSegment(
+    textSegment: ContentModelText,
+    parent: ContentModelParagraph,
+    start: number,
+    end: number
+): ContentModelText {
+    const text = textSegment.text;
+    const index = parent.segments.indexOf(textSegment);
+    const textBefore = createText(
+        text.substring(0, start),
+        textSegment.format,
+        textSegment.link,
+        textSegment.code
+    );
+    const middleText = createText(
+        text.substring(start, end),
+        textSegment.format,
+        textSegment.link,
+        textSegment.code
+    );
+    const textAfter = createText(
+        text.substring(end),
+        textSegment.format,
+        textSegment.link,
+        textSegment.code
+    );
+    textBefore.isSelected = textSegment.isSelected;
+    middleText.isSelected = textSegment.isSelected;
+    textAfter.isSelected = textSegment.isSelected;
+
+    parent.segments.splice(index, 1, textBefore);
```

Comment:
better check if the new segment is empty
and `splice` support adding multiple item together. So you can do this:

```ts
const newSegments: ContentModelText = [ 
createText(...) // the middle segment here
];

if (start > 0) {
  newSegments.unshift(createText(...) // first segment);
}

if (end < length - 1) {
  newSegment.push(createText(...) // last segment);
}

newSegments.forEach(x => x.isSelected = text.isSelected);
parent.segments.splice(index, 1, ...newSegments);
```